AI recipe UI

sunshinemoua opened the pull request:
- Created new page for AI assistant
- Appended specific instructions to user's request to AI chatbot
- Added button to save AI generated recipe to db and render recipe card

sunshinemoua committed:
Add Create AI Recipe page and integrate AI chatbot functionality

- Introduced a new page for creating AI-generated recipes.
- Implemented a chat interface for user interaction with the AI chatbot.
- Updated the sidebar to include a link to the new Create AI Recipe page.

sunshinemoua committed:
Enhance AI Recipe Creation with User Message Processing and UI Improvements

- Append recipe-specific requirements to last message for Chat API.
- Improved the UI of the Create AI Recipe page, including a new "Add to Recipes" button for assistant messages.

sunshinemoua committed:
Save AI recipe to db

- Updated message content requirements to include title and time specifications.
- Implemented recipe data extraction from user input, including title, description, ingredients, and instructions.
- Added user authentication check before allowing recipe submission, with error handling and success notifications.

sunshinemoua committed:
Remove unused AI chatbot functionality from DashboardContent component

gurkirtheerey commented on app/api/chat/route.ts:
```diff
@@ -21,10 +21,32 @@ export async function POST(req: Request) {
+      role,
+      content,
+    }));
+    console.log('cleanMessages', cleanMessages);
```
delete

gurkirtheerey reviewed (commented):
1. spacing for user avatar
2. margins over input
3. add to recipes will open recipe in a new tab
4. hide add to recipes if prompt doesn't meet requirement
5. fix prompt appending prep time
6. submit button rounded full
7. once added to recipes, hide button

sunshinemoua committed:
fixed styling for user avatar, submit button, and margins over input

sunshinemoua committed:
Enhance recipe request handling in chat API and improve UI feedback

- Added logic to identify explicit recipe creation requests in user messages and append specific requirements.
- Updated success notification on recipe addition to direct users to the Recipes section.
- Adjusted message display to preserve whitespace for better formatting.

sunshinemoua committed:
Refactor recipe handling in Create AI Recipe page

- Moved recipe data extraction logic to a new utility file for better organization.
- Enhanced the UI to include a "Saved" indicator for previously saved recipes.
- Updated the recipe addition handler to manage saved responses more effectively.
